lib/internal/loader/CreateDynamicModule.js

```diff
@@ -33,27 +34,41 @@ const createDynamicModule = (exports, url = '', evaluate) => {
   reflectiveModule.instantiate();
   const { setExecutor, reflect } = reflectiveModule.evaluate()();
   // public exposed ESM
+  const importLookup = {};
+  for (const name of imports)
+    importLookup[name] = Buffer.from(name).toString('base64');
+
```

Comment:
What the name decoding 👆 about?

Reply:
the names can be paths/urls which don't make very nice identifiers in js

Reply:
Would you post an example please?

Reply:
https://github.com/nodejs/node/pull/18972/files#diff-b0fe7f1fb1b686a8b721a7e5b0518b13R4

Reply:
Ahhhh so then it becomes
```js
import * as someBase64Thing from "./add.mjs"
```
Then you capture that `someBase64Thing` namespace object and assign it to another object `{ "./add.mjs": someBase64Thing }` and pass it along to get resolved and then imported into it at `new WebAssembly.Instance(module, reflect.imports)`.

Reply:
yep